compiler/tflchef/tflite/src/RecipeChef.cpp

```diff
@@ -74,254 +74,276 @@ std::unique_ptr<ModelRecipe> generate_recipe(const tflite::Model *model)
-            {
-              chef_array_indices->add_dim(as);
-            }
-            break;
-          default:
-            throw std::runtime_error("unsupported sparse index vector type");
+          auto chef_dm = chef_sparsity->add_dim_metadata();
+          // format
+          chef_dm->set_format(as_tflchef_sparse_dim_type(dm->format()));
+          // dense_size
+          chef_dm->set_dense_size(dm->dense_size());
+          // array_segments
+          auto chef_array_segments = chef_dm->mutable_array_segments();
+          switch (dm->array_segments_type())
+          {
+            case tflite::SparseIndexVector_NONE:
+              // DO NOTHING
+              break;
+            case tflite::SparseIndexVector_Int32Vector:
+              for (const auto &as : *(dm->array_segments_as_Int32Vector()->values()))
+              {
+                chef_array_segments->add_dim(as);
+              }
+              break;
+            case tflite::SparseIndexVector_Uint16Vector:
+              for (const auto &as : *(dm->array_segments_as_Uint16Vector()->values()))
+              {
+                chef_array_segments->add_dim(as);
+              }
+              break;
+            case tflite::SparseIndexVector_Uint8Vector:
+              for (const auto &as : *(dm->array_segments_as_Uint8Vector()->values()))
+              {
+                chef_array_segments->add_dim(as);
+              }
+              break;
+            default:
+              throw std::runtime_error("unsupported sparse index vector type");
+          }
+          // array_indices
+          auto chef_array_indices = chef_dm->mutable_array_indices();
+          switch (dm->array_indices_type())
+          {
+            case tflite::SparseIndexVector_NONE:
+              // DO NOTHING
+              break;
+            case tflite::SparseIndexVector_Int32Vector:
+              for (const auto &as : *(dm->array_indices_as_Int32Vector()->values()))
+              {
+                chef_array_indices->add_dim(as);
+              }
+              break;
+            case tflite::SparseIndexVector_Uint16Vector:
+              for (const auto &as : *(dm->array_indices_as_Uint16Vector()->values()))
+              {
+                chef_array_indices->add_dim(as);
+              }
+              break;
+            case tflite::SparseIndexVector_Uint8Vector:
+              for (const auto &as : *(dm->array_indices_as_Uint8Vector()->values()))
+              {
+                chef_array_indices->add_dim(as);
+              }
+              break;
+            default:
+              throw std::runtime_error("unsupported sparse index vector type");
+          }
+        }
+      }
+
+      auto shape_signature = tensor->shape_signature();
+      if (shape_signature != nullptr)
+      {
+        tflchef::ShapeSignature *chef_shape_signature = operand->mutable_shape_signature();
+        for (uint32_t j = 0; j < shape_signature->size(); ++j)
```

Comment:
Q) why is this changed to `j` ?

Reply:
Because `i` is already used in above `for` loop:
```c++
for (uint32_t i = 0; i < tensors->size(); ++i)
```

Reply:
> Because i is already used in above for loop:

Then why is the loop before was OK but now nested?

Reply:
It's a bit hard to read the changes even with `Hide whitespace`

Reply:
> > Because i is already used in above for loop:
> 
> Then why is the loop before was OK but now nested?

Before changes this loop was nested too, and it was like `shadows a local variable` problem.

Reply:
Ah, got it, thanks! Can you fire a PR for this change in separate PR ?